extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/binder/KafkaBinderProcessor.java

```diff
@@ -8,16 +8,21 @@
 import io.quarkus.micrometer.runtime.config.MicrometerConfig;
 
 /**
- * Add support for Kafka Producer and Consumer instrumentation. Note that
+ * Add support for Kafka Producer, Consumer and Streams instrumentation. Note that
  * various bits of support may not be present at deploy time. Avoid referencing
  * classes that in turn import optional dependencies.
  */
 public class KafkaBinderProcessor {
     static final String KAFKA_CONSUMER_CLASS_NAME = "org.apache.kafka.clients.consumer.Consumer";
     static final Class<?> KAFKA_CONSUMER_CLASS_CLASS = MicrometerRecorder.getClassForName(KAFKA_CONSUMER_CLASS_NAME);
 
+    static final String KAFKA_STREAMS_CLASS_NAME = "org.apache.kafka.streams.KafkaStreams";
```

Comment:
This isn't knowledge about the API so much as it is a "is this enabled" so it can happen automatically. Because we're using OOTB Micrometer binders...
There is definitely a tug of war here, and it could go either way. The way we managed it for kafka generally is that the micrometer extension knows about the binders, and knows enough to turn them on when present.

Reply:
For myself, I prefer to see a more of the metrics behavior stay with the metrics extension (as it tends to act as a unit with consistent configuration behavior).. Having metrics spread out all over everywhere I find bothersome, especially as the footprint can be kept to a minimum, and the dependencies are not transitive.

Reply:
I see, and as you say there's reasons for both ways.

I think I'm leaning towards KS -> Metrics/Health as it's done like that for Health already, and it's inline with the open/closed principle which I generally favour: when adding new "business" extensions like Kafka Streams, you don't need to "register" them in the (cross-cutting concern) Metrics extension, but you can do everything via build items in that extension itself. Same would go for external extensions we don't have any knowledge about at all.

Reply:
it's less "registering" and more "understanding that micrometer has a meter binder that you should enable". The way we're doing this (for kafka producer/consumer as well) allows kafka to emit events. That micrometer understands events and sets up the right binders in response is up to micrometer.

Reply:
Ok, going to resolve this one after quick discussio with @ebullient. We can go with this and always revisit later on if needed.

Reply:
Gasp, seems I lack rights to resolve? In any case, it is, from my PoV.